argparse_test.go

```diff
@@ -1179,3 +1180,81 @@ func TestUsageStringer(t *testing.T) {
 		t.Errorf("%s", usage)
 	}
 }
+
+func TestNewParserHelpFuncDefault(t *testing.T) {
+	parser := NewParser("parser", "")
+	if parser.HelpFunc == nil || parser.Help(nil) != parser.Usage(nil) {
+		t.Errorf("HelpFunc should default to Usage function")
+	}
+}
+
+func ExampleCommand_Help() {
+	parser := NewParser("parser", "")
+	parser.HelpFunc = func(c Command, msg interface{}) string {
+		return fmt.Sprintf("Name: %s\n", c.GetName())
+	}
+	fmt.Println(parser.Help(nil))
+
+	// Output:
+	// Name: parser
+}
+
+func ExampleCommand_Help_subcommand_defaulting() {
+	parser := NewParser("parser", "")
+	parser.HelpFunc = func(c Command, msg interface{}) string {
+		helpString := fmt.Sprintf("Name: %s\n", c.GetName())
+		for _, com := range c.GetCommands() {
+			// Calls parser.HelpFunc, because command.HelpFuncs are nil
+			helpString += com.Help(nil)
+		}
+		return helpString
+	}
+	parser.NewCommand("subcommand1", "")
+	parser.NewCommand("subcommand2", "")
+	fmt.Println(parser.Help(nil))
+
+	// Output:
+	// Name: parser
+	// Name: subcommand1
+	// Name: subcommand2
+}
+func ExampleCommand_Help_subcommand_helpfuncs() {
```

Comment:
Snake case -> Camel case

Reply:
Also if this is not a test -- it should not be in test file, if it is a test, it should begin with `Test`, if it is a test helper -- should be clear from the name that this is helper and what it does (which it currently is not clear). Same for other methods above.

Reply:
Examples are golang tests which focus on standard output, see the examples section [here](https://golang.org/pkg/testing/). The function name follows the naming convention described in the same section. If preferred, I can either move example tests to their own file, or convert them to Tests.

Reply:
Naming conventions for examples require the underscores.

> Multiple example functions for a package/type/function/method may be provided by appending a distinct suffix to the name. The suffix must start with a lower-case letter.

> func Example_suffix() { ... }
> func ExampleF_suffix() { ... }
> func ExampleT_suffix() { ... }
> func ExampleT_M_suffix() { ... }

go vet only likes "ExampleCommand_Help_subcommandHelpFuncs"

Reply:
Got it, thanks. Then that's fine. Perhaps just moving them to a separate file should be enough.